sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -306,31 +310,102 @@ export class ManagementClient extends LinkEntity {
     options?: { retryOptions?: RetryOptions; abortSignal?: AbortSignalLike; requestName?: string }
   ): Promise<any> {
     try {
-      log.mgmt(
-        "[%s] Acquiring lock to get the management req res link.",
-        this._context.connectionId
-      );
-      await defaultLock.acquire(this.managementLock, () => {
-        return this._init();
-      });
-
       if (!options) {
         options = {};
       }
 
-      const sendRequestOptions: SendRequestOptions = {
-        maxRetries: options.retryOptions && options.retryOptions.maxRetries,
-        abortSignal: options.abortSignal,
-        requestName: options.requestName,
-        timeoutInSeconds: getRetryAttemptTimeoutInMs(options.retryOptions) / 1000,
-        delayInSeconds:
-          options.retryOptions &&
-          options.retryOptions.retryInterval &&
-          options.retryOptions.retryInterval >= 0
-            ? options.retryOptions.retryInterval / 1000
-            : undefined
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+          try {
+            let count = 0;
+
+            if (!options) {
+              options = {};
+            }
+
+            const retryTimeoutInMs = getRetryAttemptTimeoutInMs(options.retryOptions);
+            let timeTakenByInit = 0;
+
+            if (!this._isMgmtRequestResponseLinkOpen()) {
+              log.mgmt(
+                "[%s] Acquiring lock to get the management req res link.",
+                this._context.connectionId
+              );
+
+              const initOperationStartTime = Date.now();
+
+              const actionAfterTimeout = () => {
+                const desc: string = `The request with message_id "${request.message_id}" timed out. Please try again later.`;
+                const e: Error = {
+                  name: "OperationTimeoutError",
+                  message: desc
+                };
+
+                return reject(translate(e));
+              };
+
+              const waitTimer = setTimeout(actionAfterTimeout, retryTimeoutInMs);
+
+              try {
+                await defaultLock.acquire(this.managementLock, () => {
+                  return this._init();
+                });
+              } catch (err) {
+                return reject(translate(err));
+              } finally {
+                clearTimeout(waitTimer);
```

Comment:
We should also check if the operation was cancelled after calling _init. If it has, we know we can close the connection and reject.

Reply:
This second check is unnecessary since we already have one at the start of `sendRequest`

Discussed offline about how `abortSignal` will need to be passed in to the `init` and other internal operations to gracefully handle cancellation _during_ init as well.
Clearing connection, links, listeners and timers will need to be done as part of this.
Since this needs to be done for receive() as well and would involve more work, changes for it and tests can be added as part of new PR and be linked to separate issue https://github.com/Azure/azure-sdk-for-js/issues/4422

Reply:
> This second check is unnecessary since we already have one at the start of sendRequest

I agree